### PR TITLE
Nerf & standardised slip times

### PR DESCRIPTION
**Pull request by K-Dynamic**
## About the PR

Nerfs a bunch of paralyze time and launch forward multipliers (increases launch speed and thus distance):

| | Paralyze time | | Forward multiplier | |
| --- | :---: | :---: | :---: | :---: |
| | Old | **New** | Old | **New** |
| Default (SlipperyComponent) | 3.0 | **1.5** | 1.0 | **1.5** |
| Puddle | 3.0 | **1.5** | 2.0 | **1.5** |
| Soap | 2.0 | **1.5** | 1.5 |  |
| Syndie soap | 5.0 | **3.0** | 3.5 | **3.0** |
| Syndie soaplets* | 5.0 | **1.5** | 2.5 | **1.5** |
| Omega soap | 7.0 | **5.0** | 3.0 |  |
| Error | 3.0 |  | 5.0 | **3.0** |
| Banana peel | 3.0 | **1.5** | 1.5 | |
| Mimana peel | 3.0 | **1.5** | 1.6 | **1.5** | 
| Bananium peel | 4.0 | **1.5** | 2.0 | **1.5** |
| Bananium tile | 2.0 | **1.5** | 1.5 | |
| Banana phone | 4.0 | **1.5** | 1.5 | |
| Space lube | 3.0 | **1.5** | 2.0 | **1.5** |
| Clown spider web | 2.0 | **1.5** | 1.5 |  |
| Clown PDA | 4.0 | **1.5** | 1.5 | |

(Blank indicates unchanged value)

*Reduced TC price of slipocalypse cluster soap from 3 to 2 TC to reflect its nerf

Unlisted items with slip components will use new default values of 1.5 paralyze time, 1.5 forward multiplier

## Why / Balance

tl;dr - slipping funny but is broken and ruins combat, short-term fix until someone codes proning and crawling system

For better or worse, stuns are a core component of combat that slipping builds on in SS13/14. It's a humorous mechanic and measure of robustness you don't see in most games, so I don't intend on nerfing it completely.

However, it's no secret slipping ranges from funny (clown slipping secoffs) to annoying (forgetting noslips and having to walk with your guns) to outright frustrating (crew and ERT newbies slipping on lubebombs and donating gear to nukies).

Slipping has several issues putting it at odds with SS14's fast-paced combat:
- Avoiding slippery objects
- Inconsistent slip times
- Unbalanced and benefits antags (even after nerf space-wizards/space-station-14#24494)

New and experienced players face visual overload and ecognising slippery objects in messy areas or stressful situations - puddles and soaps can be hidden under airlocks, shrouded in darkness, or placed mid-conversation beneath you. Certain puddles have 3-second paralyze times yet can intermix with non-slip ones. Assuming perfect focus and colour recognition by most players, the colourblind are still punished as they can't identify puddles tile-by-tile in reasonable time. High-ping players or players with jammed inputs (#1326) are also punished by slips because lag prevents reacting in time.

Slip times are also inconsistent. For instance, objects with SlipperyComponent (usually puddles and banana peels) have 3-second paralyze times, while soap has a 2-second paralyze time. Although some variation is fine, this leads to disjointed experiences or setbacks in 'timing' common slips. Only antag or rare items should have higher slip times, albeit lowered to avoid hoarding and unfair advantages as explained later.

Antags nominally report requiring noslips to avoid slipping and losing their weapon. However, this has become weaponised in combat: security officers are unable to keep pace with suspects carrying lube or soaps, and nukies craft lubebombs to disable a majority of the crew. (Security should ideally equip magboots in such scenarios, but science is not guaranteed to craft them.) Combat can be redefined around slipping, with not much difference between two-hit batons (#20970) and syndie soap without low ping and perfect walk toggles. 

In response to all my previous points, **I've nerfed slip times across the table for every object in-game.** Rare or antag items are also nerfed but will still stun longer compared to common slip items. Slipocalypse Cluster Soap price is reduced from 3 to 2 TC as each soaplet stuns shorter than Syndie Soap.

Although I've seen points about being valid or item confiscation, this isn't usually the case in practice - the clown can simply remove their ID before using their PDA to slip people, and antags either pick up or purchase new slippery items.

My PR is not perfect and leaves several issues:
- There are better long-term solutions like a prone system or launching slippery objects when slipped
   - Takes longer to code and is beyond my skill ceiling
- Proposed times and forward multipliers are arbitrary
   - Subject to feedback and further tests by players
- May tilt combat balance towards crew
- Default walk bind kinda sucks
- People can still be stunlocked by lubebombs
- Slippery objects still lack telegraphing or are easily hidden

## Technical details

cs changes:
- SlipperyComponent.cs (default slip paralyze time and launch forward multiplier)

yaml changes:
- puddle.yml (puddles)
- produce.yml (banana, mimana and bananium peel)
- pda.yml (clown PDA)
- instruments_misc.yml (banana phone)
- error.yml (error object)
- spider_web.yml (clown spiderweb)
- soap.yml (soaps)
- bananium.yml (bananium tile)
- cleaning.yml (space lube)
- uplink_catalog.yml (reduced tc price of slipocalypse cluster soap from 3 to 2 tc)

## Media

Before:

https://github.com/user-attachments/assets/8efa5da8-a4e3-4625-8781-96f661a48efd

After:

https://github.com/user-attachments/assets/da42b051-bcb1-4c81-8e40-4cd4411925e5

- [x] I have added screenshots/videos to this PR showcasing its changes ingame

## Breaking changes

N/A

**Changelog**

:cl:
- tweak: nerfed paraylze timer of all slippable objects (including soaps, water puddles, and clown-related items)

### CODE REVIEW

**Comment by ninruB** [comment]
It's important to note that `SlipperyComponent` defaults to the following relevant values:
`ParalyzeTime = 3`
`LaunchForwardsMultiplier = 1`

Slippery objects and their altered values as follows:
```
SpaceLube
    paralyzeTime: 3
    launchForwardsMultiplier: 2

FloorBananiumEntity
    paralyzeTime: 2
    launchForwardsMultiplier: 1.5

Error
    componentDefault (3)
    launchForwardsMultiplier: 5

Soap
    paralyzeTime: 2
    launchForwardsMultiplier: 1.5

SoapSyndie
    paralyzeTime: 5
    launchForwardsMultiplier: 2.5

SoapletSyndie
    paralyzeTime: 5
    launchForwardsMultiplier: 2.5

SoapOmega
    paralyzeTime: 7
    launchForwardsMultiplier: 3

Puddle
    componentDefault (3)
    launchForwardsMultiplier: 2.0

BananaPhoneInstrument
    paralyzeTime: 4
    launchForwardsMultiplier: 1.5

SpiderWebClown
      paralyzeTime: 2
      launchForwardsMultiplier: 1.5

ClownPDA
    paralyzeTime: 4
    launchForwardsMultiplier: 1.5

MobHonkBot
    componentDefault (3)
    launchForwardsMultiplier: 2

Foam
    componentDefault (3)
    componentDefault (1)

Ethanol, Laughter, Razorium, Fresium, WeldingFuel, BaseDrink, possibly more.
    componentDefault (3)
    componentDefault (1)

TrashBananaPeel
    componentDefault (3)
    launchForwardsMultiplier: 1.5

TrashMimanaPeel
    componentDefault (3)
    launchForwardsMultiplier: 1.6

TrashBananiumPeel
    paralyzeTime: 4
    launchForwardsMultiplier: 2

WhoopieCushion has SlipperyComponent with a ParalyzeTime of 0, probably used as a workaround to make sound on step?
ShardBase, d4Dice have SlipperyComponent with a launchForwardsMultiplier of 0, probably used for their "stun-on-step if shoes aren't worn" functionality.
Water has SlipperyComponent but parents from BaseDrink so that might be redundant?
```
will edit to feature more in a bit hold on

**Comment by K-Dynamic** [comment]
no idea what the test failure is about

**Comment by Hmeister-real** [comment]
Syndie soap and omega soap should remain untouched, they are meant to be big slips and nerfing them nerfs peaceful syndies too (and lucky tiders)

**Comment by Dutch-VanDerLinde** [comment]
Slipping should work like how it does in ss13 rather than just hardstunning you

**Comment by K-Dynamic** [comment]
> Slipping should work like how it does in ss13 rather than just hardstunning you

Yeah a crawling and prone system would be great

Except it disarms you anyway and you're forced to crawl rather than be propelled forwards and stunned for a bit

Plus mirror was trying to get opinions on halving stun times so I just decided to submit it

**Comment by keronshb** [comment]
> Slipping should work like how it does in ss13 rather than just hardstunning you

This requires a lot of work because you need:
- A way to shove yourself up, which requires some sort of laying/prone system as well
- Stamina drain from slipping (not hard to implement, but we need a better visual implementation of stamina)
- A discussion if we want stamina drain from slipping
- A softstun system (to prevent actions during the initial slip)

**Comment by K-Dynamic** [comment]
> This requires a lot of work because you need:
> 
> * A way to shove yourself up, which requires some sort of laying/prone system as well
> * Stamina drain from slipping (not hard to implement, but we need a better visual implementation of stamina)
> * A discussion if we want stamina drain from slipping
> * A softstun system (to prevent actions during the initial slip)

I think what they're getting at is that this pr is a bit of a band-aid, and I do kinda agree with that - I would personally rework the whole system

At the same time I'm limited in my own ability to code and time to learn, and it might be that halving slip times might be acceptable or 'good enough' - it's not like we'd forget about adding proning later. Plus there are some legitimate issues like the fact default slip times are 3 seconds, which is a lot for a game with low TTKs.

**Comment by Dysh3s** [comment]
Merge it right now, do it

**Comment by ninruB** [comment]
This is a rebalance, I don't think the feature freeze applies to tweaks, which a rebalance is.

**Comment by ninruB** [comment]
Overall it'd be better to just make slips balanced by copying 13 and implementing the mechanics Keron mentioned, but I don't think this change works as a bandaid either.

I personally think that the changes should be (*if* they should) the following:

Normal soap should be 2 seconds of stun and a launch coefficient of 1, they are the "common" stun but less common than peels.

Banana peels should be 1 second of stun and a launch coefficient of 2, this makes them less viable in combat overall but a better option for shenanigans as they launch you further.

Water puddle stun time from 1, and leave their launch coefficient at 2, this should make them less annoying overall, which is good considering *how* annoying they are.

Syndie soap either remaining at 5, or going down to 4 with a lower launch coefficient. It's a traitor item that's easy to lose and makes you valid.

Syndie soaplets from 5 seconds of stun to 1.5, launch coefficient at 1.5 as well, maybe decrease the parent item's price if that was to go through.

Omega soap stun time from 7 to 4 or 5, it's commonly a mapped item or a thief set item, 7 seconds of stun is bizarre considering that it still follows slipping rules and thus can technically be instant. Launch coefficient untouched.

Space lube stun time to 2, launch coefficient remains at 2. This ensures soap parity and makes lube a little less good, which is fair considering that it's one of the few stuns that bypass speed requirements.

Bananium tiles being at 1.5 is fine but they're a stupidly uncommon item, I don't think changing them from 2 really matters.

Banana phone and Clown PDA should remain at their values as they are rare items and typically limited to the clown, a job that is infamous for slipping people. Besides, the risk of losing the PDA should be rewarded by making the slip actually worth it.

**Comment by thebadman4662** [comment]
> Bananium peels should probably be untouched, they're not common and it makes sense for them to be an upgrade.

While its true getting one requires bananium ore, once you get some of it you get whole 10 pretty much 0TC peels that do not make you valid like syndie soap, so im not really sure how to feel about that one...

**Comment by ninruB** [comment]
> While its true getting one requires bananium ore, once you get some of it you get whole 10 pretty much 0TC peels that do not make you valid like syndie soap, so im not really sure how to feel about that one...

You can probably nerf them to being 3 stuntime and 2.5 launch coefficient, I don't think it's too big of a deal if it becomes a recurring issue in rounds.

**Comment by AlexUm418** [comment]
if it's getting merged i will go and cry in a pillow

**Comment by EmoGarbage404** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopened after this.

**Comment by K-Dynamic** [comment]
Reopen

**Comment by metalgearsloth** [comment]
Provide a video please.

**Comment by K-Dynamic** [comment]
> Provide a video please.

Willdo after examining feedback in comments and updating the PR

**Comment by metalgearsloth** [comment]
I'm fine with nerfing these short-term regardless of whether someone wants to redesign slips / stuns in future I just want to see it ingame.

**Comment by superjj18** [comment]
> > Slipping should work like how it does in ss13 rather than just hardstunning you
> 
> 
> 
> Yeah a crawling and prone system would be great
> 
> 
> 
> Except it disarms you anyway and you're forced to crawl rather than be propelled forwards and stunned for a bit
> 
> 
> 
> Plus mirror was trying to get opinions on halving stun times so I just decided to submit it

A crawling system will likely be needed eventually anyways, especially with newmed allowing for legless characters 


I do love that puddle slips aren't insta-game over in combat situations tho

**Comment by K-Dynamic** [comment]
Converting to draft while I update this PR (holy moly my description was misleading), take in some feedback and provide a video

**Comment by retequizzle** [comment]
we should make omega soap have an even faster forward multiplier so they can get launched into space

(changes look good otherwise, should make slips less oppressive in standard gameplay without taking away the entirety of their usage cases for players)

**Comment by K-Dynamic** [comment]
One of the reasons for this PR was because we lacked crawling. It's worth taking a look at the values again since https://github.com/space-wizards/space-station-14/pull/36881 has been merged.
